python/paddle/fluid/dygraph/dygraph_to_static/loop_transformer.py

```diff
@@ -62,6 +64,54 @@ def create_while_node(condition_name, body_name, loop_var_names):
     return assign_node
 
 
+class LogicalOpTransformer(gast.NodeTransformer):
+    """
+    Transform python boolean op into Paddle logical op
+    """
+
+    def __init__(self, node):
+        self.root = node
+
+    def transform(self):
+        return self.visit(self.root)
+
+    def visit_UnaryOp(self, node):
+        self.generic_visit(node)
+        if isinstance(node.op, gast.Not):
+            arg = ast_to_source_code(node.operand)
+            new_node_str = "fluid.layers.logical_not({})".format(arg)
+            # gast.parse returns Module(body=[expr(value=...)])
+            new_node = gast.parse(new_node_str).body[0].value
+            return new_node
+
+    def visit_BoolOp(self, node):
+        self.generic_visit(node)
+        if isinstance(node.op, gast.And):
+            new_node = self._create_bool_op_node(node.values, 'and')
+        elif isinstance(node.op, gast.Or):
+            new_node = self._create_bool_op_node(node.values, 'or')
+        else:
+            raise TypeError(
+                "Only supports and/or syntax in control flow if statement.")
+        return new_node
+
+    def _create_bool_op_node(self, nodes, api_type):
+        assert len(
+            nodes
+        ) > 1, "The length of BoolOp should be at least 2, but received {}.".format(
+            len(nodes))
+        if len(nodes) > 2:
+            # Creates logic_and/logic_or node recursively.
+            pre_assign_node = _create_node(nodes[:2], api_type)
+            nodes = [pre_assign_node] + nodes[2:]
```

Comment:
It seems that`_create_node` is not defined or imported here.

In `NodeTestTransformer`, I added  `_create_node` function to be called recursively because if `BoolOp` was `A and B and C`，the `node.values` will be `values=[A, B, C]` with `length=3`. However, `logical_and` only receives two params (`x` and `y`). So I transformed them recursively.

Reply:
Here, if you just want to transform `A and B`, you can assert `len(nodes) == 2`, and remove other unused code.

Reply:
Yes, I found it when adding test cases and I have done the change